pyqmc/linemin.py

```diff
@@ -136,16 +136,29 @@ def line_minimization(
         warmup_options["tstep"] = vmcoptions["tstep"]
     assert npts >= 3, f"linemin npts={npts}; need npts >= 3 for correlated sampling"
 
-    # Restart
     iteration_offset = 0
-    if hdf_file is not None and os.path.isfile(hdf_file):
+    if hdf_file is not None and os.path.isfile(hdf_file): #restarting -- read in data
         with h5py.File(hdf_file, "r") as hdf:
             if "wf" in hdf.keys():
                 grp = hdf["wf"]
                 for k in grp.keys():
                     wf.parameters[k] = gpu.cp.asarray(grp[k])
             if "iteration" in hdf.keys():
                 iteration_offset = np.max(hdf["iteration"][...]) + 1
+            coords.load_hdf(hdf)
```

Comment:
What happens here if the number of configurations changes? It looks like it will just silently change: 
```
        """Note that the number of configurations will change to reflect the number in the hdf file."""
        # The ... seems to be necessary to avoid changing the dtype and screwing up
        # pyscf's calls.
        self.configs[...] = np.array(hdf["configs"])
```

Reply:
From will:  self.configs[...] = hdf["configs"][()] should error out if the shape of the array is different, because of the …